### PR TITLE
V button color prop

### PR DESCRIPTION
**Pull request by jamescammarano**
Added `info`, `danger`, `warning`, and `success` style vars to VButton.
Usage: add var to class

added to `archive-action` buttons

gif

### CODE REVIEW

File: app/src/modules/collections/routes/collection.vue [hunk 1]
```diff
@@ -127,7 +127,7 @@
 						<v-button secondary @click="confirmArchive = false">
 							{{ t('cancel') }}
 						</v-button>
-						<v-button class="action-archive" :loading="archiving" @click="archive">
+						<v-button class="action-archive warning" :loading="archiving" @click="archive">
```
Comment (rijkvanzanten) [inline app/src/modules/collections/routes/collection.vue:130]: I'm wondering if we should align this with `v-notice` 🤔 

`v-notice` uses `type` prop for the same purpose. That wouldn't work for `v-button` though as type is already an existing attribute on `<button>` elements.

Comment (rijkvanzanten) [reply app/src/modules/collections/routes/collection.vue:130]: Ref:



https://github.com/directus/directus/blob/0b17fdcf8163d82edd146e84c90bc2dafd6a9bde/app/src/components/v-notice/v-notice.vue#L13-L16

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

File: app/src/modules/collections/routes/collection.vue (hunk 1 above)
Comment (jaycammarano) [reply app/src/modules/collections/routes/collection.vue:130]: I was thinking the same thing about the type property already being used. I could name it something like "style-type" on VButton.

Comment (rijkvanzanten) [reply app/src/modules/collections/routes/collection.vue:130]: In lieu of `style` / `type`, lets use `kind` as the prop name, feels fairly natural: `<v-button kind="danger">` 

cc @benhaynes

Comment (jaycammarano) [reply app/src/modules/collections/routes/collection.vue]: fixed in 34684b1

File: app/src/components/v-button/v-button.vue [hunk 2]
```diff
@@ -165,7 +170,24 @@ export default defineComponent({
 			return false;
 		});
 
-		return { sizeClass, onClick, component, isActiveRoute, toggle };
+		const isKind = computed(() => {
```
Comment (rijkvanzanten) [inline app/src/components/v-button/v-button.vue:173]: I don't think this is used/needed

File: app/src/modules/collections/routes/collection.vue [hunk 3]
```diff
@@ -143,7 +144,7 @@
 							<v-button secondary @click="confirmArchive = false">
 								{{ t('cancel') }}
 							</v-button>
-							<v-button class="action-archive" :loading="archiving" @click="archive">
+							<v-button class="action-archive warning" :loading="archiving" @click="archive">
```
Comment (rijkvanzanten) [inline app/src/modules/collections/routes/collection.vue:147]: Replace warning class with `kind="warning"`

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]
We'll have to go over the codebase, and replace every button color override with the kind prop

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**